sqlglot/dialects.py

```diff
@@ -209,31 +209,98 @@ def _format_time(exp_class, dialect, default=None):
+        f"STRFTIME({MIXED_TYPE_TO_DATE_EXPRESSION}, {DATE_FORMAT})"
+    )
+    MIXED_TYPE_TO_DI_EXPRESSION = (
+        "CAST(SUBSTR(REPLACE(CAST({this} AS VARCHAR), '-', ''), 1, 8) AS INT)"
+    )
 
-    def _ts_or_ds_add(self, expression):
-        this = self.sql(expression, "this")
-        e = self.sql(expression, "expression")
-        unit = self.sql(expression, "unit").strip("'") or "DAY"
+    @classmethod
+    def _mixed_type_add_to_di(cls, generator, expression):
+        this = generator.sql(expression, "this")
+        e = generator.sql(expression, "expression")
+        unit = expression.text("unit") or "DAY"
+
+        date = DuckDB.MIXED_TYPE_TO_DATE_EXPRESSION.format(this=this)
+
+        return f"CAST(STRFTIME({date} + INTERVAL {e} {unit}, {DuckDB.DATEINT_FORMAT}) AS INT)"
+
+    @classmethod
+    def _mixed_type_add_to_ds(cls, generator, expression):
+        this = generator.sql(expression, "this")
+        e = generator.sql(expression, "expression")
+        unit = expression.text("unit") or "DAY"
         return f"STRFTIME(CAST({this} AS DATE) + INTERVAL {e} {unit}, {DuckDB.DATE_FORMAT})"
 
+    def _unix_to_time(self, expression):
+        return f"TO_TIMESTAMP(CAST({self.sql(expression, 'this')} AS BIGINT))"
+
     def _date_add(self, expression):
         this = self.sql(expression, "this")
         e = self.sql(expression, "expression")
-        unit = self.sql(expression, "unit").strip("'") or "DAY"
+        unit = expression.text("unit") or "DAY"
         return f"{this} + INTERVAL {e} {unit}"
 
+    def _mixed_type_to_date_str_sql(self, expression):
+        this = self.sql(expression, "this")
+        return DuckDB.MIXED_TYPE_TO_DATE_STR_EXPRESSION.format(this=this)
+
+    def _mixed_type_to_di_sql(self, expression):
+        this = self.sql(expression, "this")
+        return DuckDB.MIXED_TYPE_TO_DI_EXPRESSION.format(this=this)
+
+    def _mixed_type_to_date_sql(self, expression):
+        this = self.sql(expression, "this")
+        return DuckDB.MIXED_TYPE_TO_DATE_EXPRESSION.format(this=this)
+
+    def _mixed_type_date_diff_sql(self, expression):
+        third_arg = self.sql(expression, "end_date")
+        if third_arg:
+            start_date_value = self.sql(expression, "start_date")
+            end_date_value = third_arg
+            unit = expression.text("unit")
+        else:
+            start_date_value = expression.text("unit")
+            end_date_value = self.sql(expression, "start_date")
+            unit = "DAY"
+
+        first_date = DuckDB.MIXED_TYPE_TO_DATE_EXPRESSION.format(this=start_date_value)
+        second_date = DuckDB.MIXED_TYPE_TO_DATE_EXPRESSION.format(this=end_date_value)
+
+        return f"DATE_DIFF('{unit}', {first_date}, {second_date})"
+
+    def _mixed_type_add_sql(self, expression):
+        output_format = expression.text("output_format").upper() or "YYYY-MM-DD"
+        if output_format == "YYYYMMDD" or isinstance(expression, exp.DiAdd):
```

Comment:
hmm, yyyymmdd can be a string "20000101", it doesn't necessitate date int right??